multigeiger/display.cpp

```diff
@@ -9,81 +9,77 @@
 
 #include "display.h"
 
-#if CPU == STICK
 #define PIN_DISPLAY_ON 25
-#endif
 
 #define PIN_OLED_RST 16
 #define PIN_OLED_SCL 15
 #define PIN_OLED_SDA 4
 
-#if CPU != STICK
 U8X8_SSD1306_128X64_NONAME_HW_I2C u8x8(PIN_OLED_RST, PIN_OLED_SCL, PIN_OLED_SDA);
-#else
-U8X8_SSD1306_64X32_NONAME_HW_I2C u8x8(PIN_OLED_RST, PIN_OLED_SCL, PIN_OLED_SDA);
-#endif
+U8X8_SSD1306_64X32_NONAME_HW_I2C u8x8_lora(PIN_OLED_RST, PIN_OLED_SCL, PIN_OLED_SDA);
+U8X8* pu8x8;
 
 bool displayIsClear;
+static bool haveLoRa;
 
 void display_start_screen(void) {
   char line[20];
 
-  u8x8.clear();
-  #if CPU == STICK
-  // Display is only 4 lines by 8 characters; lines counting from 2 to 5
-  u8x8.setFont(u8x8_font_5x8_f);        // use really small font
-  for (int i = 2; i < 6; i++) {
-    u8x8.drawString(0, i, "        ");  // clear all 4 lines
+  pu8x8->clear();
+  if(haveLoRa) {
+    // Display is only 4 lines by 8 characters; lines counting from 2 to 5
+    pu8x8->setFont(u8x8_font_5x8_f);        // use really small font
+    for (int i = 2; i < 6; i++) {
+      pu8x8->drawString(0, i, "        ");  // clear all 4 lines
+    }
   }
-  #endif
-
-  #if CPU == STICK
-  u8x8.drawString(0, 2, "Geiger-");
-  u8x8.drawString(0, 3, " Counter");
-  u8x8.drawString(0, 4, "Version:");
-  snprintf(line, 9, "%s", VERSION_STR);  // 8 chars + \0 termination
-  u8x8.drawString(0, 5, line);
-  #else
-  u8x8.setFont(u8x8_font_7x14_1x2_f);
-  u8x8.drawString(0, 0, "Geiger-Counter");
-  u8x8.drawString(0, 2, "==============");
-  snprintf(line, 15, "%s", VERSION_STR);  // 14 chars + \0 termination
-  u8x8.drawString(0, 4, line);
-  u8x8.drawString(0, 6, "Info:boehri.de");
-  #endif
-
+ 
+ if(haveLoRa) {
+    pu8x8->drawString(0, 2, "Geiger-");
+    pu8x8->drawString(0, 3, " Counter");
+    pu8x8->drawString(0, 4, "Version:");
+    snprintf(line, 9, "%s", VERSION_STR);  // 8 chars + \0 termination
+    pu8x8->drawString(0, 5, line);
+ } else {
+    pu8x8->setFont(u8x8_font_7x14_1x2_f);
+    pu8x8->drawString(0, 0, "Geiger-Counter");
+    pu8x8->drawString(0, 2, "==============");
+    snprintf(line, 15, "%s", VERSION_STR);  // 14 chars + \0 termination
+    pu8x8->drawString(0, 4, line);
+    pu8x8->drawString(0, 6, "Info:boehri.de");
+}
   displayIsClear = false;
 };
 
-void setup_display(void) {
-  u8x8.begin();
-
-  #if CPU == STICK
-  pinMode(PIN_DISPLAY_ON, OUTPUT);
-  digitalWrite(PIN_DISPLAY_ON, HIGH);
-  #endif
-
+void setup_display(bool lora) {
+  haveLoRa = lora;
```

Comment:
as far as this module is concerned, it is not really about having lora or not, it is about which kind of display we have.
which is also implied if we know which `board` we have.